broker/src/main/java/org/apache/rocketmq/broker/metrics/BrokerMetricsManager.java

```diff
@@ -459,9 +462,14 @@ private void initConnectionMetrics() {
             .ofLongs()
             .buildWithCallback(measurement -> {
                 Map<ProducerAttr, Integer> metricsMap = new HashMap<>();
-                brokerController.getProducerManager()
-                    .getGroupChannelTable()
-                    .values()
+                ConcurrentHashMap<String, ConcurrentHashMap<Channel, ClientChannelInfo>> groupChannelTable = brokerController.getProducerManager().getGroupChannelTable();
+                ConcurrentHashMap<String, ConcurrentHashMap<Channel, ClientChannelInfo>> clone = new ConcurrentHashMap<>(groupChannelTable);
+                clone.forEach((group, map) -> {
+                    if (MixAll.CLIENT_INNER_PRODUCER_GROUP.equals(group)) {
+                        groupChannelTable.remove(group);
```

Comment:
looks that you removed the keys in `groupChannelTable` but used clone map at last